test/data/test_builtin_datasets.py

```diff
@@ -126,3 +126,22 @@ def test_imdb(self):
         conditional_remove(datafile)
         datafile = os.path.join(self.project_root, ".data", "aclImdb_v1.tar.gz")
         conditional_remove(datafile)
+
+    @slow
+    def test_iterabledataset_imdb(self):
+        from torchtext.experimental.datasets import IMDB
+        # smoke test to ensure imdb works properly
+        train_dataset, = IMDB(data_select='train', raw_text=True)
+        cached_dataset = list(train_dataset)
```

Comment:
If you're materializing the entire dataset you could also compare it to a sort-of hacky version of IMDB. We know the layout of this particular dataset and can simply read it in via read(<path>).read().split('\n'). Then we could do an actual comparison to an underlying reference (for now).